packages/console/src/pages/Applications/components/GetStarted/index.tsx

```diff
@@ -20,10 +24,57 @@ type Props = {
   onClose: () => void;
 };
 
+type DocumentFileNames = {
+  files: string[];
+};
+
+type Step = {
+  title?: string;
+  subtitle?: string;
+  metadata: string;
+};
+
 const GetStarted = ({ appName, onClose }: Props) => {
   const [isLibrarySelectorFolded, setIsLibrarySelectorFolded] = useState(false);
-  const [libraryName, setLibraryName] = useState<string>(SupportedJavascriptLibraries.Angular);
+  const [libraryName, setLibraryName] = useState<string>(SupportedJavascriptLibraries.React);
+  const [steps, setSteps] = useState<Step[]>();
+  const [activeStepIndex, setActiveStepIndex] = useState<number>(-1);
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const publicPath = useMemo(
+    () => `/console/get-started/${libraryName}/${i18next.language}`,
+    [libraryName]
+  );
+  const stepReferences = useMemo(
+    () => Array.from({ length: steps?.length ?? 0 }).map(() => React.createRef<HTMLDivElement>()),
+    [steps?.length]
+  );
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await fetch(`${publicPath}/index.json`);
```

Comment:
We can use `useSWR`:
```ts
const { data } = useSWR<SomeType[]>(urls, async (urls) => new Promise.all(xxxx));
```

And if we take another step by using middleware, the `urls` and `files` can be fetch by a single line of `useSWR` code.

Reply:
But I think swr will re-fetch the data when I switch back and forth between browser tabs, right? I would not like to reload these markdown files as they won't change whatsoever

Reply:
Is swr's job to manage the revalidation, just like many other APIs we used with GET method.

Reply:
Btw, in this situation, I believe it will not refetch.

Reply:
Well, I will give it a try and update the result

Reply:
Your concerns are covered by `useSWR`

Reply:
Updated, ptal